crates/control/src/behavior/intercept_ball.rs

```diff
@@ -1,16 +1,39 @@
 use nalgebra::{point, Point2, UnitComplex};
+use spl_network_messages::{GamePhase, SubState};
 use types::{
-    configuration::InterceptBall, HeadMotion, LineSegment, MotionCommand, OrientationMode,
-    PathSegment, PrimaryState, WorldState,
+    configuration::InterceptBall, FilteredGameState, GameControllerState, HeadMotion, LineSegment,
+    MotionCommand, OrientationMode, PathSegment, WorldState,
 };
 
 pub fn execute(world_state: &WorldState, parameters: InterceptBall) -> Option<MotionCommand> {
     match (
-        world_state.robot.primary_state,
+        world_state.filtered_game_state,
         world_state.ball,
         world_state.robot.robot_to_field,
+        world_state.game_controller_state,
     ) {
-        (PrimaryState::Playing, Some(ball), Some(robot_to_field)) => {
+        (
+            Some(FilteredGameState::Playing { .. }),
```

Comment:
I think if we are in any of the two game phases we do not want interception behavior independent of the `FilteredGameState`. If so, you don't have to match on `Playing` here:
```suggestion
            _,
```

Reply:
Good catch, changed